src/components/LanguagePicker/index.tsx

```diff
@@ -0,0 +1,226 @@
+
+import MenuItem from "./MenuItem"
+import NoResultsCallout from "./NoResultsCallout"
+import { useLanguagePicker } from "./useLanguagePicker"
+
+type LanguagePickerProps = Omit<MenuListProps, "children"> & {
+  children: React.ReactNode
+  placement?: MenuProps["placement"]
+  handleClose?: () => void
+  menuState?: UseDisclosureReturn
+}
+
+const LanguagePicker = ({
+  children,
+  placement,
+  handleClose,
+  menuState,
+  ...props
+}: LanguagePickerProps) => {
+  const { t, refs, disclosure, filterValue, setFilterValue, filteredNames } =
+    useLanguagePicker(handleClose, menuState)
+  const { inputRef, firstItemRef, noResultsRef, footerRef } = refs
+  const { onClose } = disclosure
+
+  /**
+   * Adds a keydown event listener to focus filter input (\).
+   * @param {string} event - The keydown event.
+   */
+  useEventListener("keydown", (e) => {
+    if (e.key !== "\\") return
+    e.preventDefault()
+    inputRef.current?.focus()
+  })
+
+  return (
+    <Menu isLazy placement={placement} autoSelect={false} {...disclosure}>
+      {children}
+      <MenuList
+        position="relative"
+        overflow="auto"
+        borderRadius="base"
+        py="0"
+        onKeyDown={(e) => {
+          if (e.key === "Tab" || e.key === "\\") {
+            e.preventDefault()
+            ;(e.shiftKey ? inputRef : footerRef).current?.focus()
+          }
+        }}
+        {...props}
+      >
+        {/* Mobile Close bar */}
+        <Flex
+          justifyContent="end"
+          hideFrom="lg" // TODO: Confirm breakpoint after nav-menu PR merged
+          position="sticky"
+          zIndex="sticky"
+          top="0"
+          bg="background.base"
+        >
+          <Button
+            p="4"
+            variant="ghost"
+            alignSelf="end"
+            onClick={() => onClose()}
+          >
+            {t("common:close")}
+          </Button>
+        </Flex>
+
+        {/* Main Language selection menu */}
+        <Box
+          position="relative"
+          w="100%"
+          minH="calc(100% - 53px)" // Fill height with space for close button on mobile
+          p="4"
+          bg="background.highlight"
+          sx={{ "[role=menuitem]": { py: "3", px: "2" } }}
+        >
+          <FormControl>
+            <FormLabel fontSize="xs" color="body.medium">
+              {t("page-languages-filter-label")}{" "}
+              <Text as="span" textTransform="lowercase">
+                ({filteredNames.length} {t("common:languages")})
+              </Text>
+            </FormLabel>
+            <InputGroup>
+              <Input
+                type="search"
+                autoComplete="off"
+                placeholder={t("page-languages-filter-placeholder")}
+                value={filterValue}
+                onChange={(e) => setFilterValue(e.target.value)}
+                onBlur={(e) => {
+                  if (e.relatedTarget?.tagName.toLowerCase() === "div") {
+                    e.currentTarget.focus()
```

Comment:
Couldn't understand what this is for, could you explain? (we could add a comment)